merino-cache/src/memory.rs

```diff
@@ -413,4 +415,41 @@ mod tests {
         LOCK_TABLE.update(other_lock_name, lock, || lock_check = false);
         assert!(lock_check);
     }
+
+    #[tokio::test]
+    async fn test_reconfigure() {
+        let cache: Arc<DedupedMap<String, Instant, Vec<Suggestion>>> = Arc::new(DedupedMap::new());
+        // Provide an inspectable metrics sink to validate the collected data.
+        let (_, sink) = SpyMetricSink::new();
+        let metrics_client = StatsdClient::from_sink("merino-test", sink);
+
+        let mut provider = Suggester {
+            inner: Box::new(NullProvider),
+            metrics_client,
+            items: cache.clone(),
+            default_ttl: Duration::from_secs(30),
+            default_lock_timeout: Duration::from_secs(1),
+            max_background_removals: usize::MAX,
+        };
+
+        // This won't be called as `DelayProvider::reconfigure()` will always succeed.
+        let make_fresh: MakeFreshType = Box::new(move |_fresh_config: SuggestionProviderConfig| {
+            ready(Ok(Box::new(NullProvider) as Box<dyn SuggestionProvider>)).boxed()
```

Comment:
Since we claim this never gets called, should we put an `assert!(false)` here?

Reply:
(if you think it would be useful asserting here, please kindly make sure to assert in the other tests as well)

Reply:
Good point! Thanks!

I ended up adding `unreachable()` as Clippy was picky on `assert!(false)`.